code/game/atoms_movable.dm

```diff
@@ -279,7 +279,10 @@
 
 /atom/movable/proc/Moved(atom/oldloc, direction, Forced = FALSE)
 	SEND_SIGNAL(src, COMSIG_MOVABLE_MOVED, oldloc, direction, Forced)
+	var/mob/living/carbon/human/H = src
 	if(pulledby)
+		if(istype(src, /mob/living/carbon/human) && istype(pulledby, /mob/living/carbon/xenomorph) && H.health < H.health_threshold_crit)
```

Comment:
you are checking the type of src, after you cast it, so it doesn't make any sense.

And in general, you never do a istype(src). This proc is way too generic, find another one. Don't put code like that in a proc that is called everytime anything move

Reply:
![image](https://user-images.githubusercontent.com/25491240/110949960-8ad20800-8343-11eb-87d7-08d445a55e97.png)

Human already have a moved proc, use that instead

Reply:
Wrong it should use signals